packages/js/test/plugins/auctionHouseModule/directSell.test.ts

```diff
@@ -0,0 +1,160 @@
+import spok, { Specifications } from 'spok';
+import { Keypair } from '@solana/web3.js';
+
+killStuckProcess();
+
+test('[auctionHouseModule] sell on an Auction House with minimum input', async (t: Test) => {
+  // Given we have an Auction House and an NFT.
+  const mx = await metaplex();
+  const buyer = await createWallet(mx);
+
+  const nft = await createNft(mx);
+  const auctionHouse = await createAuctionHouse(mx);
+
+  // And we put a public bid on that NFT for 1 SOL.
+  const { bid } = await mx
+    .auctionHouse()
+    .bid({
+      auctionHouse,
+      buyer,
+      mintAccount: nft.address,
+      price: sol(1),
+      seller: mx.identity().publicKey,
+    })
+    .run();
+
+  // Then we execute an Sell on the bid
+  const { purchase } = await mx
+    .auctionHouse()
+    .sell({ auctionHouse, bid })
+    .run();
+
+  // Then we created and returned the new Purchase with appropriate values.
+  const expectedPurchase = {
+    price: spokSameAmount(sol(1)),
+    tokens: spokSameAmount(token(1)),
+    buyerAddress: spokSamePubkey(buyer.publicKey),
+    sellerAddress: spokSamePubkey(mx.identity().publicKey),
+    auctionHouse: {
+      address: spokSamePubkey(auctionHouse.address),
+    },
+    asset: {
+      address: spokSamePubkey(nft.address),
+      token: {
+        address: findAssociatedTokenAccountPda(nft.address, buyer.publicKey),
+        ownerAddress: spokSamePubkey(buyer.publicKey),
+      },
+    },
+    receiptAddress: spok.defined,
+  };
+  spok(t, purchase, {
+    $topic: 'Purchase',
+    ...expectedPurchase,
+  } as unknown as Specifications<Purchase>);
+});
+
+test('[auctionHouseModule] sell on an Auction House with auctioneer', async (t: Test) => {
+  // Given we have an Auction House and an NFT.
+  const mx = await metaplex();
+  const buyer = await createWallet(mx);
+
+  const nft = await createNft(mx);
+  const auctioneerAuthority = Keypair.generate();
+  const auctionHouse = await createAuctionHouse(mx, auctioneerAuthority);
+
+  // And we put a public bid on that NFT for 1 SOL.
+  const { bid } = await mx
+    .auctionHouse()
+    .bid({
+      auctionHouse,
+      buyer,
+      mintAccount: nft.address,
+      price: sol(1),
+      seller: mx.identity().publicKey,
+      auctioneerAuthority,
+    })
+    .run();
+
+  // Then we execute an Sell on the bid
+  const { purchase } = await mx
+    .auctionHouse()
+    .sell({ auctionHouse, bid, auctioneerAuthority })
+    .run();
+
+  // Then we created and returned the new Purchase
+  t.equal(purchase.asset.address.toBase58(), nft.address.toBase58());
+});
+
+test('[auctionHouseModule] sell on an Auction House with maximum input', async (t: Test) => {
+  // Given we have an Auction House and an NFT.
+  const mx = await metaplex();
+  const buyer = await createWallet(mx);
+  const seller = await createWallet(mx);
+
+  const nft = await createNft(mx, { tokenOwner: seller.publicKey });
+  const auctionHouse = await createAuctionHouse(mx, null, {
+    authority: seller,
```

Comment:
@zaxozhu When you create tests for "maximum input" scenarios, please create a new Keypair for every single parameter unless they have to be the same.

Here, there should be another Keypair for the authority of the auction house.

Also use `Keypair.generate()` instead of `createWallet` whenever you can so that:
- A. the tests are a bit faster
- B. we make sure that no lamports are required from the former otherwise an error will be thrown.